### ISSUE TITLE
Notifications aren't working for me

### ISSUE THREAD

**Issue by Cassolotl** [glitch-soc/mastodon#815]
I have a Pixel 2 running Android 9. My browser is "Free Adblock Browser", which is based on Chrome I think?

My instance is dev.glitch.social. I've added it as an icon with "add to home screen" via the browser. In the notifications settings, I have every toggle under "mentions" switched on.

When I do this with mastodon.social I get notifications. When I do it with dev.glitch.social I don't get notifications.

I'm not sure what else to tell you so please do ask questions if I can help to narrow it down or in case I've done something wrong!

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bugs also occur on vanilla Mastodon

**Comment by ThibG** [comment]
I can confirm I can't get push notifications to work with “Free Adblocker Browser” on dev.glitch.social (haven't tried another instance yet). I'm not sure why for now: it works on desktop Firefox.

Normally, you should have 4 settings per kind of notification, but the web-push related one does not show up in “Free Adblocker Browser” (it doesn't work on Firefox Mobile either, but that one doesn't support web push at all afaik)

**Comment by Cassolotl** [comment]
I was going to ask whether I should report this problem to the browser devs (whoever they are??), but then I do get notifications with mastodon.social so it can't be the browser's fault!

These are my settings anyway:

![photo_2018-11-15_18-11-26](https://user-images.githubusercontent.com/6607512/48572648-e28e0f00-e901-11e8-9712-b17d17fd1b12.jpg)

**Comment by ThibG** [comment]
I don't have a mastodon.social account to test, but I have tried from another Mastodon instance running upstream code, and I don't have notifications either.
From a quick look at how it works, it seems it doesn't handle VAPID notifications at all, so push notifications should not work at all?

**Comment by Cassolotl** [comment]
Well, then I am confused, because I definitely got notifications to work with Free Adblock Browser and mastodon.social! Let me see if I can get a screenshot.

![screenshot_20181116-131732](https://user-images.githubusercontent.com/6607512/48623706-5a137b00-e9a2-11e8-9304-5147efd6c389.png)

**Comment by ThibG** [comment]
Weird. I don't have a mastodon.social account, I'm not sure why “Free Adblocker Browser” would work with it but not on other vanilla instances I tried :/

**Comment by Cassolotl** [comment]
I don't know what else I could try to narrow down the bug! :S

**Comment by Cassolotl** [comment]
I dug around in all the settings I could find (they were in three places!) and deleted all the permissions for dev.glitch.social, then I logged out and deleted the PWA from my home screen, then I restarted my phone for good measure, then I did the whole process again and accepted notification permissions and added as a PWA, and it worked? I guess it is just a process that involves a lot of steps in obscure places and I hadn't managed to catch them all on the other attempts!

So, I've got notifications working for dev.glitch.social now and everything is good, I will go ahead and close this. I'm sorry for making extra work for you unnecessarily!
